### PR TITLE
Use 2 components in code example for child-parent communication

### PR DESCRIPTION
**Pull request by jshawl**
The current code sample only references one component, which makes it difficult to understand how I might communicate between two components.

### CODE REVIEW

**Comment by facebook-github-bot-9** [comment]
Thank you for your pull request and welcome to our community. We require contributors to sign our Contributor License Agreement, and we don't seem to have you on file. In order for us to review and merge your code, please sign up at https://code.facebook.com/cla - and if you have received this in error or have any questions, please drop us a line at cla@fb.com. Thanks!

**Comment by zpao** [comment]
Sorry for the delay. Overall this seems fine but we need to make some more changes for it to fit more naturally. Comments incoming.

File: docs/tips/14-communicate-between-components.md [hunk 1]
```diff
@@ -10,30 +10,40 @@ next: expose-component-functions.html
...
-  console.log('You clicked: ' + props.items[i]);
-}
+var GroceryList = React.createClass({
+  show: function(e){
+    console.log(e);
+  },
+  render: function(){
+    return(
+      <div>
+        {this.props.items.map(function(item, index){
```
Comment (zpao) [inline docs/tips/14-communicate-between-components.md]: Please match our style in regards to spacing around brackets and parents.

File: docs/tips/14-communicate-between-components.md [hunk 2]
```diff
@@ -10,30 +10,40 @@ next: expose-component-functions.html
...
+          return <Item key={index} value={item} log={this.show}/>
+        }.bind(this))}
```
Comment (zpao) [inline docs/tips/14-communicate-between-components.md]: Please don't use `.bind(this)` here. `map()` takes an extra argument for scope, or you can pass an arrow function to `map()`

File: docs/tips/14-communicate-between-components.md [hunk 3]
```diff
@@ -10,30 +10,40 @@ next: expose-component-functions.html
...
+  }
+})
 
 ReactDOM.render(
   <GroceryList items={['Apple', 'Banana', 'Cranberry']} />, mountNode
 );
 ```
 
-Notice the use of `bind(this, arg1, arg2, ...)`: we're simply passing more arguments to `handleClick`. This is not a new React concept; it's just JavaScript.
+Notice the use of `bind(this)`: we're ensuring `this` inside the `map()` refers to the `GroceryList`. This is not a new React concept; it's just JavaScript.
```
Comment (zpao) [inline docs/tips/14-communicate-between-components.md]: This whole line really goes away with the new code. We can replace this first sentence with something discussing the same topic - that we're passing a bound function through, even though we don't use bind.

Comment (jshawl) [inline docs/tips/14-communicate-between-components.md]: how about:

> Notice the use of the arrow function - `=>`: we're ensuring `this` inside the `map()` refers to the `GroceryList`. This is not a new React concept; it's just JavaScript.

File: docs/tips/14-communicate-between-components.md [hunk 4]
```diff
@@ -10,30 +10,40 @@ next: expose-component-functions.html
...
+  }
+})
 
 ReactDOM.render(
   <GroceryList items={['Apple', 'Banana', 'Cranberry']} />, mountNode
 );
 ```
 
-Notice the use of `bind(this, arg1, arg2, ...)`: we're simply passing more arguments to `handleClick`. This is not a new React concept; it's just JavaScript.
+Notice the use of the arrow function `=>`: we're ensuring `this` inside the `map()` refers to the `GroceryList`. This is not a new React concept; it's just JavaScript.
```
Comment (zpao) [inline docs/tips/14-communicate-between-components.md]: Sounds good, just a couple additions to make it flow a little better (but not actually emphasized).

> `this` inside the `map()` **_call_** refers to the `GroceryList` **_component_**.

Comment (jshawl) [inline docs/tips/14-communicate-between-components.md]: 👍 Would you like me to rebase and push?

**Comment by gaearon** [comment]
Sorry for delays in reviewing. We completely replaced this doc so it shouldn't be relevant anymore.

**Pull request closed without merging**